packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -170,6 +170,11 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
         callClient.offStateChange(onStateChange);
         return;
       }
+      // Always use current call.id value to access client state
+      // Since call.id could be change during the call
+      if (this.call?.id) {
```

Comment:
does this mean whenever we have a state change and there is a call.id then we should set the callId? This sounds like it would potentially fire way more often and cause even more re-renders than only when the id has changed?